### DISCUSSION TITLE
New level: Privacy II

### DISCUSSION THREAD

**Discussion by Asamartino** [OpenZeppelin/ethernaut#513] [Level idea]
Hi,

I have an idea for another Ethernaut level. It will be in continuation of the Privacy level. Except this time one part of the password will be hidden in a dynamic array and the other part in a mapping.
To solve this level the user will have to understand how data is stored in dynamically-sized values and compute the location of it in order to retrieve it (as explained in [this article](https://programtheblockchain.com/posts/2018/03/09/understanding-ethereum-smart-contract-storage/)).

You can find a first draft of the proposed challenge [here](https://github.com/Asamartino/EthernautChallenges/blob/master/PrivacyII_proposal.sol) ( pragma ^0.6.7)

What do you think ?

**Comment by abcoathup** [comment]
Hi @Asamartino,

I'm sorry that I haven't gotten back to you on this yet.  
I like the idea.  

I am thinking of the best way to upgrade to Solidity 0.6 and create a Solidity 0.6 version of Ethernaut.

**Comment by dahifi** [comment]
@abcoathup, I don't think the Delegation challenge will work in 0.6.0 since they updated the way fallback functions work. 

In the process of downgrading my compiler and SafeMath libs to 0.5.0 so I can play along.
